csrc/quantization/fp8/common.cuh

```diff
@@ -32,25 +61,35 @@ __device__ __forceinline__ float atomicMaxFloat(float* addr, float value) {
   return old;
 }
 
-template <bool is_scale_inverted>
-__device__ __forceinline__ FP8_TYPE scaled_fp8_conversion(float const val,
-                                                          float const scale) {
+template <bool is_scale_inverted, typename fp8_type>
+__device__ __forceinline__ void scaled_fp8_conversion(fp8_type& ret,
```

Comment:
Why the update by reference instead of returning the value?

Reply:
I added `fp8_type` as a template param and I panicked that you can't have two functions that differ in just the return type in C++.

Reply:
Turns out this works just fine the way it was before and with fp8_type as a template param.  Will revert.